File: 프로그래머스/lv1/131705. 삼총사/삼총사.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int g_allcnt = 0;

void selection(vector<int> number, int idx=0, int allsum=0, int cnt=0) {
    if (cnt == 3) {
        if (allsum == 0)
            g_allcnt++;
        return;
    }
    if (idx >= number.size())
        return;
    selection(number, idx+1, allsum + number[idx], cnt+1);
    selection(number, idx+1, allsum, cnt);
    return;
}

int solution(vector<int> number) {
    selection(number);
    return g_allcnt;
}

File: 프로그래머스/lv1/131705. 삼총사/삼총사.py
def selection(number, idx=0, allsum=0, cnt=0):
    global g_allcnt
    if (cnt == 3):
        if (allsum == 0):
            g_allcnt += 1
        return
    if (idx >= len(number)):
        return
    selection(number, idx+1, allsum + number[idx], cnt+1)
    selection(number, idx+1, allsum, cnt)
    return


def solution(number):
    global g_allcnt
    g_allcnt = 0
    selection(number)
    return g_allcnt